### PR TITLE
Addition: Auto Noise Removal (Batch Script)

### LINKED ISSUE #947: Automated Noise Removal
# Description
The fundamental aim of this script is to automate the process of Audio Extraction, Profiling and Cleaning before the Mixing Stage for Voice-Overs or Audio Recordings in videos that is adjustable according to the user using Python. This is achieved with the help of the following open-sourced software:
FFmpeg
Sox
Audacity

## Language
- [x] Python

| name            | about                                                  | labels  | assignees                                                      |
| Automated Noise Removal | Extract Audio from multiple video files and de-noise them automatically | Python, NewFunctionality, opensource | @owlrana |

**Is your feature request related to a problem? Please describe.**: 

I'm always frustrated when I need to denoise a number of audio files at once. One needs to first extract each audio file from videos, then create a noise profile separately for each audio file and export them separately after this process.
The whole job takes me more than an hour for 5-7 audio files.  Below is the solution that I propose to this problem.

**Describe the solution you'd like**: 
The solution is to automate the process of extraction, creating a noise profile for each audio file separately using python and then de-noise it using Audacity's already developed open algorithms, then also export it. 
This should take no more than 2-3 minutes for 10+ audio files, which will be a huge improvement from 1 hour.

### PR DESCRIPTION
**Pull request by owlrana**
# Description

Added automated functionality Audio Extraction, Profiling and Cleaning (De-Noise) through this script.

Fixes #947

## Type of change

_Please delete options that are not relevant._

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by owlrana** [comment]
I have used "tabs --> 4 spaces" settings in Visual Studio Code. Is there any other way I can reduce the issues shown?

File: Python/Automated_Noise_Removal/AutomatedNoiseRemoval.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
+"""
```
Comment (vybhav72954) [inline Python/Automated_Noise_Removal/AutomatedNoiseRemoval.py:1]: ALl this detail will go into `README`

File: Python/Automated_Noise_Removal/AutomatedNoiseRemoval.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+
+import sys
+import os
+import glob
+import subprocess
+from pydub import AudioSegment
+
+
+MYSOX_PATH = '"C:\\Users\\Rahul Rana\\OneDrive - Microsoft Student Ambassadors\\Azure Applications\\sox\\sox.exe"' # IS DIFFERENT FOR EACH SYSTEM!! NEED TO INSTALL SOX FIRST!!
```
Comment (vybhav72954) [inline Python/Automated_Noise_Removal/AutomatedNoiseRemoval.py:23]: Remember to add links to SOX.

And python hates inline comments, please put the comment in `l22`

File: Python/Automated_Noise_Removal/AutomatedNoiseRemoval.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+		
+		if "denoiseCall" in operations:
+			denoiseCall(f)
+	
+	if "cleanDir" in operations:
+		cleanDir()
+
+
+def denoiseCall(fileInput):	
+
```
Comment (vybhav72954) [inline Python/Automated_Noise_Removal/AutomatedNoiseRemoval.py:55]: A blank line before doscstring is not required

File: Python/Automated_Noise_Removal/AutomatedNoiseRemoval.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+	
+		"root" : sys.argv[1],
+		"blankCall" : None 
+		# [SAME] Last file does not have to do with anything!
+	
+	}
+
+	performOperations(operations, parameters)
+	
+	sys.exit(0)
```
Comment (vybhav72954) [inline Python/Automated_Noise_Removal/AutomatedNoiseRemoval.py:155]: blank line at EOF missing

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

**Comment by owlrana** [comment]
@vybhav72954 thanks for the assistance!
Changes are done, please review :')

File: Python/Automated_Noise_Removal/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+OPEN-SOURCE SCRIPT FOR AUTOMATED NOISE REMOVAL USING PYTHON.
```
Comment (vybhav72954) [inline Python/Automated_Noise_Removal/README.md]: Hey it would be great if you can follow the template we use.

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

File: Python/Automated_Noise_Removal/README.md (hunk 5 above)
Comment (owlrana) [reply Python/Automated_Noise_Removal/README.md]: @vybhav72954 Just did! Please check and let me know if any further changes are required.
Have a good day!

File: Python/Automated_Noise_Removal/README.md [hunk 6]
```diff
@@ -1,9 +1,41 @@
...
+- Python
 
-If you have any questions/queries related to this script you can contact me at: me.rahulrana@outlook.com
\ No newline at end of file
+## Setup instructions
+
+There is no installation for script to work on your system. Just download the script and run on any machine that has python 3+. The only packages that you might need to install would be:
+1. SoX - ( http://sox.sourceforge.net/ )
+2. pydub - ( https://pypi.org/project/pydub/ )
+3. Python 3+ (if not already installed) - ( https://www.python.org/downloads/ )
```
Comment (vybhav72954) [inline Python/Automated_Noise_Removal/README.md:13]: We don't need python3 here :P

File: Python/Automated_Noise_Removal/README.md [hunk 7]
```diff
@@ -1,9 +1,41 @@
...
+- FFmpeg
+- Sox
+- Python
 
-If you have any questions/queries related to this script you can contact me at: me.rahulrana@outlook.com
\ No newline at end of file
+## Setup instructions
+
+There is no installation for script to work on your system. Just download the script and run on any machine that has python 3+. The only packages that you might need to install would be:
+1. SoX - ( http://sox.sourceforge.net/ )
```
Comment (vybhav72954) [inline Python/Automated_Noise_Removal/README.md:11]: Would be great if you can hyperlink these urls

File: Python/Automated_Noise_Removal/README.md [hunk 8]
```diff
@@ -1,9 +1,41 @@
...
+2. pydub - ( https://pypi.org/project/pydub/ )
```
Comment (vybhav72954) [inline Python/Automated_Noise_Removal/README.md:12]: The pip command should be here as well

File: Python/Automated_Noise_Removal/README.md [hunk 9]
```diff
@@ -1,9 +1,41 @@
...
+
+
+
+After Running:
+
+![alt text](https://i.imgur.com/VjEH8TA.png)
+
+## Author
+
+Rahul Rana (@owlrana)
```
Comment (vybhav72954) [inline Python/Automated_Noise_Removal/README.md:37]: Hyperlinks would be a better idea.

File: Python/Automated_Noise_Removal/README.md [hunk 10]
```diff
@@ -1,9 +1,41 @@
-OPEN-SOURCE SCRIPT FOR AUTOMATED NOISE REMOVAL USING PYTHON
+IGNORE THIS CHECK VERSION
```
Comment (vybhav72954) [inline Python/Automated_Noise_Removal/README.md]: Is this a placeholder or something? I don't think we need this line

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

**Comment by owlrana** [comment]
@vybhav72954 Changes made :')

**Comment by owlrana** [comment]
@vybhav72954 Hi, is there anything else to be done or can we merge for now?

**Comment by owlrana** [comment]
@vybhav72954 is there any other feature you would like for me to add?

**Comment by owlrana** [comment]
@vybhav72954 Can I get an update on this?

**Comment by owlrana** [comment]
@vybhav72954 Please merge the pull request.

**Comment by owlrana** [comment]
@vybhav72954  Are we ready to merge?

**Review by vybhav72954 — APPROVED** [review]
LGTM

**Pull request merged**